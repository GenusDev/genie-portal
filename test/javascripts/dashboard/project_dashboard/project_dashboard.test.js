import React from 'react';
import ProjectDashboard from 'Entities/dashboard/project_dashboard/project_dashboard';
import { shallow, mount } from 'enzyme';

function setUp(){
  const props = {
    fetchProjects: jest.fn(() => {
      return Promise.resolve();
    }),
    currentUser: {
      email: "matsteele@gmail.com",
      password: "password"
    }
  };

  const eWrapper = shallow(<ProjectDashboard {...props}/>);
  const mountWrapper = mount(<ProjectDashboard {...props}/>);

  mountWrapper.setState({
    projectClicked: {
      title: "Project 1",
      summary: "Project Summary",
      model_id: "1",
      cashflow: projectCashflow
    },
    showText: true,
    openModal: true
  });

  return {
    props,
    eWrapper,
    mountWrapper
  };

}

describe('ProjectDashboard Component', () => {
  it('should render without throwing an error', () => {
    const { eWrapper } = setUp();
    expect(eWrapper).toMatchSnapshot();
  });

  // it('should have a summary', () => {
  //   const { mountWrapper } = setUp();
  //
  //   let summaryDiv = mountWrapper.find('div.project-summary')
  //   console.log(mountWrapper.props().currentUser);
  //   expect(summaryDiv.text()).toBeTruthy();
  // })

  it('should have a cashflow graph', () => {

  });

  it('should have a Project Thermo', () => {

  });

  it('should have a leaflet map', () => {

  });

});

let projectCashflow = `{
  "01A": -36974,
  "02A": -40018,
  "03A": -16857,
  "04A": -2915,
  "05A": -20325,
  "06A": 7864,
  "07A": 25360,
  "08A": 28107,
  "09A": 28942,
  "10A": 28696,
  "11A": 29356,
  "12A": 28854,
  "13A": 28588,
  "14A": 30781,
  "15A": 29081,
  "16A": 31887,
  "17A": 51887,
  "18A": 71887,
  "19P": 30339,
  "20P": 30718,
  "21P": 31102,
  "22P": 31491,
  "23P": 31885,
  "24P": 32283,
  "25P": 32687,
  "26P": 33096,
  "27P": 33509,
  "28P": 33928
}`;
