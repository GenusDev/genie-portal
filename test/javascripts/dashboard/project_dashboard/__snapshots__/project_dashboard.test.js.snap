// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectDashboard Component should render without throwing an error 1`] = `
<div
  className="graph-container"
>
  <ProjectGraph
    chart="loading"
    closeModal={[Function]}
    currentUser={
      Object {
        "email": "matsteele@gmail.com",
        "password": "password",
      }
    }
    fetchProjects={
      [MockFunction] {
        "calls": Array [
          Array [],
          Array [],
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": Promise {},
          },
          Object {
            "isThrow": false,
            "value": Promise {},
          },
          Object {
            "isThrow": false,
            "value": Promise {},
          },
        ],
      }
    }
    openModal={[Function]}
  />
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="modal-container"
    closeTimeoutMS={0}
    contentLabel="Project Graph Modal"
    isOpen={false}
    onRequestClose={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "borderRadius": "10px",
          "bottom": "5%",
          "color": "white",
          "height": "80%",
          "left": "10%",
          "padding": "0px",
          "position": "fixed",
          "right": "5%",
          "top": "5%",
          "width": "80%",
          "zIndex": 101,
        },
        "overlay": Object {
          "backgroundColor": "rgba(255, 255, 255, 0.75)",
          "bottom": 0,
          "left": 0,
          "position": "fixed",
          "right": 0,
          "top": 0,
          "zIndex": 10,
        },
      }
    }
  >
    <div
      className="black-close-modal-button close-modal-button"
      onClick={[Function]}
    >
      Ã—
    </div>
    <h1
      className="nodata-text"
    >
      No data available
    </h1>
  </Modal>
</div>
`;
